src/js/components/Button/stories/Badge.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+
+import { Box, Button, grommet, Grommet } from 'grommet';
+import { Alert, Notification, SettingsOption } from 'grommet-icons';
+
+export const Badge = props => (
+  <Grommet theme={grommet}>
+    <Box direction="row" align="center" pad="medium" gap="medium">
+      <Button
+        aria-label="Label, 2 Alerts"
+        label="Label"
+        onClick={() => {}}
+        badge={2}
+        secondary
+        {...props}
+      />
+      <Button
+        aria-label="100 Unread Notifications"
+        label="Notifications"
+        onClick={() => {}}
+        badge={{
+          background: 'status-warning',
+          value: 100,
+          max: 9,
+        }}
+        secondary
+        {...props}
+      />
+      <Button
+        aria-label="Alert on Settings"
+        icon={<SettingsOption />}
+        onClick={() => {}}
+        badge={
+          <Box background="status-critical" pad="xsmall" round="xsmall">
+            <Alert size="small" />
+          </Box>
+        }
+        hoverIndicator
+        {...props}
+      />
+      <Button
+        aria-label="23 Unread Notifications"
+        icon={<Notification />}
```

Comment:
Similar comment to @ShimiSun about positioning on icon only buttons is odd, but don't currently have any immediate recommendations, unless we were to conditionally move the Stack inside of the styledButtonResult or set the guiding child differently.

Reply:
Took a new approach here now that allows caller to specify target: 'contents' to indicate they want the offset relative to the button contents as opposed to the outer edges of the button.